packages/core/src/fhircast/index.ts

```diff
@@ -131,66 +143,136 @@ export type SubscriptionRequest = {
+export type FhircastSyncErrorContext = FhircastOperationOutcomeContext;
+
+export type FhircastHubContentContext = {
+  key: 'content';
+  resource: Bundle;
 };
 
+// Type utility to get keys for a specific event
+export type FhircastEventKeys<EventName extends FhircastEventName> = keyof (typeof FHIRCAST_EVENT_RESOURCES)[EventName];
+
+// Type utility to extract the resource type from an event and key
+export type FhircastContextResourceType<
+  EventName extends FhircastEventName,
+  K extends FhircastEventKeys<EventName>,
+> = (typeof FHIRCAST_EVENT_RESOURCES)[EventName][K] extends { resourceType: infer R } ? R : never;
+
+export type FhircastEventContext<EventName extends FhircastEventName = FhircastResourceEventName> =
+  EventName extends 'Patient-open'
+    ? FhircastPatientOpenContext
+    : EventName extends 'Patient-close'
+      ? FhircastPatientCloseContext
+      : EventName extends 'ImagingStudy-open'
+        ? FhircastImagingStudyOpenContext
+        : EventName extends 'ImagingStudy-close'
+          ? FhircastImagingStudyCloseContext
+          : EventName extends 'Encounter-open'
+            ? FhircastEncounterOpenContext
+            : EventName extends 'Encounter-close'
+              ? FhircastEncounterCloseContext
+              : EventName extends 'DiagnosticReport-open'
+                ? FhircastDiagnosticReportOpenContext
+                : EventName extends 'DiagnosticReport-close'
+                  ? FhircastDiagnosticReportCloseContext
+                  : EventName extends 'DiagnosticReport-update'
+                    ? FhircastDiagnosticReportUpdateContext
+                    : EventName extends 'DiagnosticReport-select'
+                      ? FhircastDiagnosticReportSelectContext
+                      : EventName extends 'syncerror'
+                        ? FhircastSyncErrorContext
+                        : never;
+
+export type FhircastPatientEvents = 'Patient-open' | 'Patient-close';
+export type FhircastImagingStudyEvents = 'ImagingStudy-open' | 'ImagingStudy-close';
+export type FhircastEncounterEvents = 'Encounter-open' | 'Encounter-close';
+export type FhircastDiagnosticReportEvents =
+  | 'DiagnosticReport-open'
+  | 'DiagnosticReport-close'
+  | 'DiagnosticReport-update'
+  | 'DiagnosticReport-select';
+export type FhircastAnchorResourceEvents =
+  | FhircastPatientEvents
+  | FhircastImagingStudyEvents
+  | FhircastEncounterEvents
+  | FhircastDiagnosticReportEvents;
+
+export type FhircastExtractAnchorResource<EventName extends FhircastAnchorResourceEvents> =
+  EventName extends FhircastPatientEvents
+    ? 'Patient'
+    : EventName extends FhircastImagingStudyEvents
+      ? 'ImagingStudy'
+      : EventName extends FhircastEncounterEvents
+        ? 'Encounter'
+        : EventName extends FhircastDiagnosticReportEvents
+          ? 'DiagnosticReport'
+          : never;
+
+export type CurrentContext<EventName extends FhircastAnchorResourceEvents | '' = FhircastAnchorResourceEvents | ''> =
```

Comment:
Made the `CurrentContext` type a union, making types more accurate based on the type of context it is, or using `''` for empty context as specified:
https://build.fhir.org/ig/HL7/fhircast-docs/2-9-GetCurrentContext.html